NTupleAnalyzerHH_nanoAOD.py

```diff
@@ -1006,50 +1077,43 @@ def LeptonsAndJetsFromHH(T):
-		#if  T.PFJetPassLooseIDAK4CHS[n]==1 and T.PFJetPtAK4CHS[n]>15 and abs(T.PFJetEtaAK4CHS[n])<2.4 : #morse only use jets that pass id #fixme loosening to study HH
-		#if  T.PFJetPassLooseIDAK4CHS[n]==1 and T.PFJetPtAK4CHS[n]>20 and abs(T.PFJetEtaAK4CHS[n])<2.4 :
+
+	daughterJets = [1,2,3,4,5,-5,-4,-3,-2,-1] #Get all quark flavors
+
+	for n in range(len(PFJetPt)):
 		if True :
 			m = TLorentzVector()
-			m.SetPtEtaPhiM(T.Jet_pt[n],T.Jet_eta[n],T.Jet_phi[n],0)
+			if radius == 4:
+				m.SetPtEtaPhiM(PFJetPt[n],PFJetEta[n],PFJetPhi[n],PFJetMass[n])
+			if radius == 8:
+				m.SetPtEtaPhiM(PFJetPt[n],PFJetEta[n],PFJetPhi[n],PFJetMass[n])
 			jetsZ.append(m)
 			jetsH.append(m)
-			recojetOriIndsZ.append(n)
+			recojetOriIndsZ.append(n) #######These are the original Reco Jets. original index because the h jets and z jets go through second selection?
 			recojetOriIndsH.append(n)
-	for n in range(T.nGenPart):
-		pdg = T.GenPart_pdgId[n]
-		if pdg not in daughterJets:
-			continue
-		motherIndex = T.GenPart_genPartIdxMother[n]
-		motherid = 0
-		if motherIndex>-1:
-			motherid = T.GenPart_pdgId[motherIndex]
-			motherStatus = T.GenPart_status[motherIndex]
-                        grandMotherid = -1
-                        if T.GenPart_genPartIdxMother[motherIndex]>-1:
-                                grandMotherid = T.GenPart_pdgId[T.GenPart_genPartIdxMother[motherIndex]]
-			#if parent=='Z' : print 'daughter:',pdg,'  mother:',motherid, '  grandMother:',grandMotherid
-			#if v=='' : print 'daughter:',pdg,'  Status:',T.GenParticleStatus[n],'  mother:',motherid,'  grandMother:',grandMotherid
-
-		if motherid not in [23,-23,25,-25] :#parentParticles:
-			continue
-		m = TLorentzVector()
-		m.SetPtEtaPhiM(T.GenPart_pt[n],T.GenPart_eta[n],T.GenPart_phi[n],0.0)
-		#if v == '':print 'Qaurk jet:',pdg,'  mother:',motherid, ' onShellZLep ', (onShellZMu or onShellZEle), '  status:',T.GenParticleStatus[n],'pt',m.Pt(),'eta',m.Eta(),'phi',m.Phi()
-		if not (onShellZMu or onShellZEle) :
-			#if v == '':print ' Not onShellZ','jet:',pdg,'  mother:',motherid,'  status:',T.GenParticleStatus[n],'pt',m.Pt(),'eta',m.Eta()
-			if motherid in [23,-23] : genjetsZ.append(m)
-			if motherid in [25,-25] : genjetsH.append(m)
-		if onShellZMu or onShellZEle :
-			#if v == '':print ' onShellZ','jet:',pdg,'  mother:',motherid,'  status:',T.GenParticleStatus[n],'pt',m.Pt(),'eta',m.Eta()
-			genjetsZH.append(m)
+
+	if isdata==False:
+
+		for n in range(T.nGenPart):
+			pdg = T.GenPart_pdgId[n]
+			if pdg not in daughterJets:
+				continue
+			motherIndex = T.GenPart_genPartIdxMother[n]
+			motherid = 0
+			if motherIndex>-1:
+				motherid = T.GenPart_pdgId[motherIndex]
+				motherStatus = T.GenPart_status[motherIndex]
+				# grandMotherid = T.GenPart_pdgId[T.GenPart_genPartIdxMother[motherIndex]]
+
+			if motherid not in [23,-23,25,-25] :#parentParticles:
+				continue
+			m = TLorentzVector()
+			m.SetPtEtaPhiM(T.GenPart_pt[n],T.GenPart_eta[n],T.GenPart_phi[n],T.GenPart_mass[n])
```

Comment:
This looks like a problem in the indentation compared to the old version, please check.